WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2504,7 +2505,8 @@ private GutenbergPropsBuilder getGutenbergPropsBuilder() {
                     postType,
                     hostAppNamespace,
                     featuredImageId,
-                    themeBundle
+                    themeBundle,
+                    isNetworkConnected
```

Comment:
My understanding of the props passed to the editor is that they are "initial props," i.e. they are never updated. Even if we did update them in the host app, I do not believe the editor would recognize updates; also, doing so might also be inefficient (re-rendering the entire editor). 

Because of this, the approach I've drafted thus far for iOS in https://github.com/WordPress/gutenberg/pull/56609 relies upon a bridge method and subscription to a bridge event. 

How do you think we should reconcile the two, currently divergent approaches? 

If you'd prefer me implement the Android side mimicking my current iOS approach, happy to do so — just let me know. Splitting the work by platform may be more difficult than I originally envisioned.

Reply:
From reviewing similar logic in the project, I came across `featuredImageId` implemented in https://github.com/WordPress/gutenberg/pull/30806 and https://github.com/WordPress/gutenberg/pull/31347. It appears to combine the approach of an [initial prop](https://github.com/WordPress/gutenberg/blob/9d43d5098b354b581e2c92c14a5cc8d38f08d43c/packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/WPAndroidGlue/GutenbergProps.kt#L42) with a [subscription](https://github.com/WordPress/gutenberg/blob/9d43d5098b354b581e2c92c14a5cc8d38f08d43c/packages/edit-post/src/editor.native.js#L101-L112) to receives updates to the value. 

The pro of an initial prop would be avoiding the need for an [asynchronous bridge request](https://github.com/WordPress/gutenberg/blob/8d7221df1876548572b208e1d63f4d7b51ea03d8/packages/react-native-bridge/index.js#L199-L212) for the initial value for each usage of the Hook. 

The cons might be:

1. A need to pass the initial value down to Hook usages, i.e. using the global store or React Context (`featuredImageId` is stored on the entity in the global store);
2. And the initial prop would be an initial "seed" value and ignored after first render, which could be (possibly necessary) indirection. 

"Taking a step back," we could place the value in the global store and hoist the subscription to the top of the editor like [many other bridge subscriptions](https://github.com/WordPress/gutenberg/blob/9d43d5098b354b581e2c92c14a5cc8d38f08d43c/packages/editor/src/components/provider/index.native.js#L110-L180), but my initial thought is that this network connection state doesn't need to be in the global store or attached to the editor package. WDYT?

Reply:
> How do you think we should reconcile the two, currently divergent approaches?

I agree that this approach was experiment, and I wasn't certain if it would actually work without implementing the connection logic subscription.

I think it would be most efficient for you to apply the same approach as in your iOS PRs. The [ConnectionChangeReceiver](https://github.com/wordpress-mobile/WordPress-Android/blob/87446592f030c7d1f0f9a6425da1bcb042c18b09/WordPress/src/main/java/org/wordpress/android/networking/ConnectionChangeReceiver.java) events observe when connectivity has changed.